frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/import-feature-flag-modal/import-feature-flag-modal.component.ts

```diff
@@ -1,46 +1,94 @@
-  imports: [CommonModalComponent, CommonModule, SharedModule, CommonImportContainerComponent],
+  imports: [CommonModalComponent, CommonModule, SharedModule, CommonImportContainerComponent, CommonStatusIndicatorChipComponent],
   templateUrl: './import-feature-flag-modal.component.html',
   styleUrls: ['./import-feature-flag-modal.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ImportFeatureFlagModalComponent {
-
+  isDescriptionExpanded = false;
+  compatibilityDescription = '';
+  displayedColumns: string[] = ['actions', 'fileName', 'compatibilityType'];
   isImportActionBtnDisabled = new BehaviorSubject<boolean>(true);
+  importFileErrorsDataSource = new MatTableDataSource<ValidateFeatureFlagError>();
+  importFileErrors: ValidateFeatureFlagError[] = [];
+  fileData: FeatureFlagFile[] = [];
+  uploadedFileCount = new BehaviorSubject<number>(0);
+  isLoadingImportFeatureFlag$ = this.featureFlagsService.isLoadingImportFeatureFlag$;
+  isImportActionBtnDisabled$: Observable<boolean> = combineLatest([
+    this.uploadedFileCount,
+    this.isLoadingImportFeatureFlag$,
+  ]).pipe(map(([uploadedCount, isLoading]) => isLoading || uploadedCount === 0));
 
   constructor(
-    @Inject(MAT_DIALOG_DATA)
-    public data: CommonModalConfig,
-    public dialog: MatDialog,
+    @Inject(MAT_DIALOG_DATA) public data: CommonModalConfig,
+    public featureFlagsService: FeatureFlagsService,
+    public featureFlagsDataService: FeatureFlagsDataService,
     public dialogRef: MatDialogRef<ImportFeatureFlagModalComponent>
   ) {}
 
-  handleFilesSelected(files: File[]) {
-    if(files.length>0) {
+  async handleFilesSelected(event) {
+    if (event.target.files.length > 0) {
       this.isImportActionBtnDisabled.next(false);
+      this.featureFlagsService.setIsLoadingImportFeatureFlag(true);
+    }
+
+    this.uploadedFileCount.next(event.target.files.length);
+    this.importFileErrors = [];
+    this.fileData = [];
+
+    if (event.target.files.length === 0) return;
+
+    const filesArray = Array.from(event.target.files) as any[];
+    await Promise.all(
+      filesArray.map((file) => {
+        return new Promise<void>((resolve) => {
+          const reader = new FileReader();
+          reader.onload = (e) => {
+            const fileContent = e.target.result;
+            this.fileData.push({ fileName: file.name, fileContent: fileContent });
+            resolve();
+          };
+          reader.readAsText(file);
+        });
+      })
+    );
+
+    await this.checkValidation(this.fileData);
+  }
+
+  async checkValidation(files: FeatureFlagFile[]) {
+    this.importFileErrors =
+      (
+        (await this.featureFlagsDataService.validateFeatureFlag(files).toPromise()) as ValidateFeatureFlagError[]
+      ).filter((data) => data.compatibilityType != null) || [];
+    this.importFileErrorsDataSource.data = this.importFileErrors;
+    this.featureFlagsService.setIsLoadingImportFeatureFlag(false);
```

Comment:
It looks like ` Observable<Object>.toPromise()` is deprecated and is replaced with `firstValueFrom` and `lastValueFrom`.  Could you update this?